### PR TITLE
feat(data-list-compact): added compact version of data list

### LINKED ISSUE #2637: Data-list: add compact variation
Spacing should match the compact table:

* 8px spacer at the top/bottom of rows
* 16px spacer between cells
* Spacing in expanded content should remain the same

cc @mcarrano @maryshak1996

### PR DESCRIPTION
**Pull request by mattnolting**
closes #2637

### CODE REVIEW

**Comment by christiemolloy** [comment]
The spacing between the first primary/secondary is a lot smaller than the 
second one, might not be anything to fix in this PR but wanted to point it out

Screen Shot 2020-02-10 at 2 46 06 PM

File: src/patternfly/components/DataList/examples/DataList.md [hunk 1]
```diff
@@ -305,7 +413,8 @@ When a list item includes more than one block of content, it can be difficult fo
 | `.pf-c-data-list__icon`  | `<i>` | Initiates a data list icon. |
 | `.pf-c-data-list__expandable-content` | `<div>` | Initiates an expandable content container. |
 | `.pf-c-data-list__expandable-content-body` | `<div>` | Initiates an expandable content container body. **Required** when `.pf-c-data-list__expandable-content` is used. |
-| `.pf-m-expanded` | `.pf-c-table__item` | Modifies for expanded state. |
+| `.pf-m-expanded` | `.pf-c-data-list__item` | Modifies for expanded state. |
+| `.pf-m-compact` | `.pf-c-data-list` | Modifies for compact state. |
```
Comment (christiemolloy) [inline src/patternfly/components/DataList/examples/DataList.md]:
```suggestion
| `.pf-m-compact` | `.pf-c-data-list` | Modifies for the compact variation. |
```

**Comment by christiemolloy** [comment]
Can you update the example so that this isn't taking up space here

Screen Shot 2020-02-10 at 2 52 08 PM

**Comment by mcoker** [comment]
@christiemolloy I put up a PR to address that hidden action cell https://github.com/patternfly/patternfly-next/pull/2691

File: src/patternfly/components/DataList/data-list.scss [hunk 2]
```diff
@@ -367,3 +387,17 @@
     overflow-y: auto;
   }
 }
+
+.pf-c-data-list.pf-m-compact {
```
Comment (christiemolloy) [inline src/patternfly/components/DataList/data-list.scss]: can this just be `&.p-m-compact` and sit inside of `pf-c-data-list`?

File: src/patternfly/components/DataList/examples/DataList.md (hunk 1 above)
Comment (christiemolloy) [inline src/patternfly/components/DataList/examples/DataList.md]: I think you had changed this to `Modifies for the compact variation`

**Pull request merged**